core/runtime/src/main/java/org/apache/isis/core/runtime/sessiontemplate/AbstractIsisSessionTemplate.java

```diff
@@ -90,46 +78,47 @@ protected void doExecuteWithTransaction(final Object context) {}
 
     // //////////////////////////////////////
 
-    protected final ObjectAdapter adapterFor(final Object targetObject) {
-        if(targetObject instanceof OidDto) {
-            final OidDto oidDto = (OidDto) targetObject;
-            return adapterFor(oidDto);
-        }
-        if(targetObject instanceof CollectionDto) {
-            final CollectionDto collectionDto = (CollectionDto) targetObject;
-            final List<ValueDto> valueDtoList = collectionDto.getValue();
-            final List<Object> pojoList = Lists.newArrayList();
-            for (final ValueDto valueDto : valueDtoList) {
-                ValueType valueType = collectionDto.getType();
-                final Object valueOrOidDto = CommonDtoUtils.getValue(valueDto, valueType);
-                // converting from adapter and back means we handle both
-                // collections of references and of values
-                final ObjectAdapter objectAdapter = adapterFor(valueOrOidDto);
-                Object pojo = objectAdapter != null ? objectAdapter.getObject() : null;
-                pojoList.add(pojo);
-            }
-            return adapterFor(pojoList);
-        }
-        if(targetObject instanceof Bookmark) {
-            final Bookmark bookmark = (Bookmark) targetObject;
-            return adapterFor(bookmark);
-        }
-        return getPersistenceSession().adapterFor(targetObject);
-    }
-
-    protected final ObjectAdapter adapterFor(final OidDto oidDto) {
-        final Bookmark bookmark = Bookmark.from(oidDto);
-        return adapterFor(bookmark);
-    }
-
-    protected final ObjectAdapter adapterFor(final Bookmark bookmark) {
-        final RootOid rootOid = RootOid.create(bookmark);
-        return adapterFor(rootOid);
-    }
-
-    protected final ObjectAdapter adapterFor(final RootOid rootOid) {
-        return getPersistenceSession().adapterFor(rootOid);
-    }
+//TODO[ISIS-1976] not used !?
```

Comment:
possibly changes behavior, need to investigate

Reply:
AbstractIsisSessionTemplate is public API, so while it might not be used within the framework, it's possible that it's used by application code.  However, let's remove it for now, see if we can simplify.  If we do need an API like this, it should probably take Bookmark (from the applib) rather than RootOid.  Also, should probably return just an Object (the domain object) rather than an ObjectAdapter - ie let's keep ObjectAdapter as an internal implementation detail and not in any of our APIs.